app/member-reminders/member-reminders-ui/src/main/kotlin/com/hedvig/android/memberreminders/ui/MemberReminderCards.kt

```diff
@@ -0,0 +1,228 @@
+package com.hedvig.android.memberreminders.ui
+
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.expandVertically
+import androidx.compose.animation.fadeIn
+import androidx.compose.animation.fadeOut
+import androidx.compose.animation.shrinkVertically
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.icons.Icons
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.hedvig.android.core.designsystem.component.button.HedvigContainedSmallButton
+import com.hedvig.android.core.designsystem.material3.onWarningContainer
+import com.hedvig.android.core.designsystem.material3.warningContainer
+import com.hedvig.android.core.designsystem.material3.warningElement
+import com.hedvig.android.core.designsystem.preview.HedvigPreview
+import com.hedvig.android.core.designsystem.theme.HedvigTheme
+import com.hedvig.android.core.icons.Hedvig
+import com.hedvig.android.core.icons.hedvig.normal.Warning
+import com.hedvig.android.core.ui.infocard.VectorInfoCard
+import com.hedvig.android.memberreminders.ApplicableMemberReminders
+import com.hedvig.android.memberreminders.UpcomingRenewal
+import com.hedvig.android.notification.permission.NotificationPermissionState
+import hedvig.resources.R
+import kotlinx.datetime.Clock
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.atStartOfDayIn
+import kotlinx.datetime.daysUntil
+import kotlinx.datetime.toLocalDateTime
+
+@Composable
+fun MemberReminderCards(
+  memberReminders: ApplicableMemberReminders,
```

Comment:
Yes, and I did actually have it as a list at first.
Then what happened is that the reminder for UpcomingRenewals itself also could have a list of reminders, so it was a list in a list.
Then as I was trying to test this and use this, I found myself having to take care of things like the list containing 2 instances of the `EnableNotifications` reminder for example, which logically makes no sense.

And then in the UI, I didn't find a good way to make the notification items come into the UI in a smooth way. If they were not there on the initial frame, once the UI state updated and had them in the list they would just pop up instantly.
When I made it animate in, it was *always* animating in, even if you went back to the previous screen which should already have them in the first frame (going back to profile from settings for example).

Then doing this approach, it allowed me to just use AnimatedVisibility around each one, so that if the first frame does not have it, it doesn't show, and when it comes in, it animates in using the AnimatedVisibility animationSpec. If however you come back to it, if it already was there, it just shows fully from the start, without doing an unnecessary animation to come in again.

It's definitely possible that there was another way to approach this which would not have this issues, but as I was playing with it this was the one that ticked all the boxes of what I was going for.

If you wanna play with this and see if you can improve it absolutely feel free to do so of course

Reply:
Thats what I thought :) in that case this solution makes sense!